dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/message/MessageImpl.java

```diff
@@ -297,5 +429,955 @@ void setContextualProperty(String key, Object v) {
             put(key, v);
```

Comment:
Not sure if this is called a lot, but for performance I think you can have this call putIfAbsent(key, v); instead of doing containsKey and put.